components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -2643,15 +2643,25 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
             throw new ResponseError(ErrorCodes.BAD_REQUEST, "organization ID must be a valid UUID");
         }
 
-        const checkAgainstDB = async (): Promise<{ team: Team; members: TeamMemberInfo[] }> => {
-            // We deliberately wrap the entiry check in try-catch, because we're using Promise.all, which rejects if any of the promises reject.
-            const team = await this.teamDB.findTeamById(teamId);
-            if (!team) {
-                // We return Permission Denied because we don't want to leak the existence, or not of the Organization.
-                throw new ResponseError(ErrorCodes.PERMISSION_DENIED, `No access to Organization ID: ${teamId}`);
-            }
+        const team = await this.teamDB.findTeamById(teamId);
+        if (!team) {
+            // We return Permission Denied because we don't want to leak the existence, or not of the Organization.
+            throw new ResponseError(ErrorCodes.PERMISSION_DENIED, `No access to Organization ID: ${teamId}`);
```

Comment:
We should do it the other way around, i.e. send 404 on a getById when the user doesn't have read permissions on a resource. IIRC we had that discussion already at some point. It is also symmetric with filtering resources in list requests

Reply:
But I also see that the current code doesn't have this contract and it makes sense to reuse it for the listTeams operation as well. So doing it in this PR seems reasonable.